src/store/actions/ApplicantActions.js

```diff
@@ -1,27 +1,70 @@
-// import axios from 'axios'
-// import axiosWithAuth from '../../utils/axiosWithAuth'
-
-// export const GET_PROFILEINFO_START = 'GET_GRANTS_START'
-// export const GET_PROFILEINFO_SUCCESS = 'GET_GRANTS_SUCCESS'
-// export const GET_PROFILEINFO_ERROR = 'GET_GRANTS_ERROR'
-
-
-// export const getProfileInfo = (applicant_id) => dispatch => {
-//   dispatch({ type: GET_PROFILEINFO_START })
-
-//   axiosWithAuth()
-//     .get(/* insert api url here */)
-//     .then((res) => {
-//       dispatch({ 
-//         type: GET_PROFILEINFO_SUCCESS, 
-//         payload: res.data
-//       })
-//     })
-//     .catch((err) => { 
-//       dispatch({ 
-//       type: GET_PROFILE_ERROR,
-//       payload: "Cannot load profile information"
-//       });
-//     });
-// }
+import axios from "axios";
+import axiosWithAuth from "../../utils/axiosWithAuth";
 
+export const GET_PROFILEINFO_START = "GET_PROFILEINFO_START";
+export const GET_PROFILEINFO_SUCCESS = "GET_PROFILEINFO_SUCCESS";
+export const GET_PROFILEINFO_FAILURE = "GET_PROFILEINFO_FAILURE";
+
+export const POST_APPLICANTGRANTS_START = "POST_APPLICANTGRANTS_START";
+export const POST_APPLICANTGRANTS_SUCCESS = "POST_APPLICANTGRANTS_SUCCESS";
+export const POST_APPLICANTGRANTS_FAILURE = "POST_APPLICANTGRANTS_FAILURE";
+
+export const GET_APPLICANTGRANTS_START = "GET_PROFILEINFO_START";
+export const GET_APPLICANTGRANTS_SUCCESS = "GET_PROFILEINFO_SUCCESS";
+export const GET_APPLICANTGRANTS_FAILURE = "GET_PROFILEINFO_FAILURE";
+
+export const getProfileInfo = applicant_id => dispatch => {
+  dispatch({ type: GET_PROFILEINFO_START });
+
+  axiosWithAuth()
```

Comment:
One of the coolest tricks I've picked up is returning the axiosWithAuth. This means when you dispatch the action you are returning a promise which allows you to chain .then()'s to the dispatched action. It's really useful for using history.push() to move the user to a new page only after a dispatched action is completed.